src/nfs/nfs_client_impl.h

```diff
@@ -24,32 +24,54 @@
  * THE SOFTWARE.
  */
 
-/*
- * Description:
- *     What is this file about?
- *
- * Revision history:
- *     xxxx-xx-xx, author, first version
- *     xxxx-xx-xx, author, fix bug about xxx
- */
 #pragma once
+
 #include <vector>
 #include <deque>
+#include <iostream>
 #include <dsn/tool-api/task_tracker.h>
 #include <dsn/tool-api/zlocks.h>
 #include <dsn/perf_counter/perf_counter_wrapper.h>
 #include <dsn/dist/nfs_node.h>
 #include <dsn/utility/defer.h>
 #include <dsn/utility/TokenBucket.h>
 #include <dsn/utility/flags.h>
-#include "nfs_client.h"
+#include <dsn/utility/optional.h>
+#include <dsn/tool-api/async_calls.h>
+
+#include "nfs_types.h"
+#include "nfs_code_definition.h"
 
 namespace dsn {
 namespace service {
 
 using TokenBucket = folly::BasicTokenBucket<std::chrono::steady_clock>;
 
-class nfs_client_impl : public ::dsn::service::nfs_client
+template <typename TCallback>
+task_ptr async_nfs_get_file_size(const get_file_size_request &request,
+                                 TCallback &&callback,
+                                 std::chrono::milliseconds timeout,
+                                 rpc_address server_addr)
+{
+    return rpc::call(server_addr,
+                     RPC_NFS_GET_FILE_SIZE,
+                     request,
+                     nullptr,
+                     std::forward<TCallback>(callback),
+                     timeout);
+}
+
+template <typename TCallback>
+task_ptr async_nfs_copy(const copy_request &request,
+                        TCallback &&callback,
+                        std::chrono::milliseconds timeout,
+                        rpc_address server_addr)
+{
+    return rpc::call(
+        server_addr, RPC_NFS_COPY, request, nullptr, std::forward<TCallback>(callback), timeout);
+}
+
```

Comment:
The APIs seem be not same with the origin(https://github.com/XiaoMi/rdsn/pull/771/files#diff-1eea91de75fadd0a10bf2538d55da66d9788516188eca733de36133898d7c8e0L107),  why not stay the same.

Reply:
Because it's now a free function not in any class. So to make it more clear, I prefixed the name with "async_nfs_" to ensure that the reader knows: 1. this api is asynchronous 2. this api belongs to the nfs module.